.github/workflows/build.yaml

```diff
@@ -258,13 +286,77 @@ jobs:
         if: matrix.config.apt_get != ''
         run: sudo apt-get update && sudo apt-get install ${{ matrix.config.apt_get }}
 
+      - name: Prepare Windows Environment
+        if: runner.os == 'Windows'
+        shell: powershell
+        run: |
+          $ErrorActionPreference = 'Stop'
+
+          # Determine VS2019 install path
+          $vsInstallPath = (Get-VSSetupInstance | `
+            Select-VSSetupInstance -Product '*' -Latest -Version '[16.0,17.0)' `
+          ).InstallationPath
+
+          $vcvarsall = "${vsInstallPath}\VC\Auxiliary\Build\vcvarsall.bat"
+          if (-not (Test-Path -Path $vcvarsall)) {
+            Out-Error "Could not find Visual Studio setup script"
+            exit 1
+          }
```

Comment:
To my knowledge this is unnecessary complexity. It's simply `C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvarsall.bat`.